pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -0,0 +1,151 @@
+package provisioning
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+type AlertRuleService struct {
+	defaultInterval int64
+	ruleStore       store.RuleStore
+	provenanceStore ProvisioningStore
+	xact            TransactionManager
+	log             log.Logger
+}
+
+func NewAlertRuleService(ruleStore store.RuleStore,
+	provenanceStore ProvisioningStore,
+	xact TransactionManager,
+	defaultInterval int64,
+	log log.Logger) *AlertRuleService {
+	return &AlertRuleService{
+		defaultInterval: defaultInterval,
+		ruleStore:       ruleStore,
+		provenanceStore: provenanceStore,
+		xact:            xact,
+		log:             log,
+	}
+}
+
+func (service *AlertRuleService) GetAlertRule(ctx context.Context, orgID int64, ruleUID string) (models.AlertRule, models.Provenance, error) {
+	query := &models.GetAlertRuleByUIDQuery{
+		OrgID: orgID,
+		UID:   ruleUID,
+	}
+	err := service.ruleStore.GetAlertRuleByUID(ctx, query)
+	if err != nil {
+		return models.AlertRule{}, models.ProvenanceNone, err
+	}
+	provenance, err := service.provenanceStore.GetProvenance(ctx, query.Result, orgID)
+	if err != nil {
+		return models.AlertRule{}, models.ProvenanceNone, err
+	}
+	return *query.Result, provenance, nil
+}
+
+func (service *AlertRuleService) CreateAlertRule(ctx context.Context, rule models.AlertRule, provenance models.Provenance) (models.AlertRule, error) {
+	if rule.UID == "" {
+		rule.UID = util.GenerateShortUID()
+	}
+	interval, err := service.ruleStore.GetRuleGroupInterval(ctx, rule.OrgID, rule.NamespaceUID, rule.RuleGroup)
+	// if the alert group does not exists we just use the default interval
```

Comment:
Is it possible to create a rule with no group?

Should this create the group if it does not exist? If `UpdateRuleGroup` below behaves like an upsert, this probably works fine, but it's not clear from the name of that method.